src/Docker/Runner.php

```diff
@@ -398,17 +425,37 @@ private function waitForStorageJobs(array $outputs)
      * @param string $configVersion
      * @param string $mode
      * @param InputTableStateList $inputTableStateList
+     * @param InputFileStateList $inputFileStateList
      * @return Output
      * @throws ClientException
      */
-    private function runComponent($jobId, $configId, $rowId, Component $component, UsageFileInterface $usageFile, DataLoaderInterface $dataLoader, WorkingDirectory $workingDirectory, StateFile $stateFile, ImageCreator $imageCreator, ConfigFile $configFile, OutputFilterInterface $outputFilter, $configVersion, $mode, InputTableStateList $inputTableStateList)
-    {
+    private function runComponent(
+        $jobId,
+        $configId,
+        $rowId,
+        Component $component,
+        UsageFileInterface $usageFile,
+        DataLoaderInterface $dataLoader,
+        WorkingDirectory $workingDirectory,
+        StateFile $stateFile,
+        ImageCreator $imageCreator,
+        ConfigFile $configFile,
+        OutputFilterInterface $outputFilter,
+        $configVersion,
+        $mode,
+        InputTableStateList $inputTableStateList,
+        InputFileStateList $inputFileStateList
+    ) {
         // initialize
         $workingDirectory->createWorkingDir();
-        $resultInputTablesState = $dataLoader->loadInputData($inputTableStateList);
+        list($resultInputTablesState, $resultInputFilesState) = $dataLoader->loadInputData(
+            $inputTableStateList,
+            $inputFileStateList
+        );
```

Comment:
I'm not really sure if this is the best way or not, but it's the most straightforward I think, without creating another object for the general storage state.  though that may be the better way to go 🤷

Reply:
a simple object would be much better (the interface definition is weird)